docs(textarea): updated docs for accessibility

nikkimk opened the pull request:
<!---
    - Following conventional commit format, provide a general summary of your changes in the title above.
    - Acceptable commit types in order of severity (high to low): feat, fix, docs, style, chore, perf, and test. Commit types are defined in PULL_REQUESTS.md.
    - For example,`type(component): general summary`
-->

## Description

Updated `packages/textarea/README.md` to align with [documentation standards](https://opensource.adobe.com/spectrum-web-components/guides/adding-component/#documentation-standards) which include accessibility.

## Motivation and context

This step will help our consumers use components accessibly. Additionally it gives the accessibility auditors, enough context to audit components.

It will ensure the full API is documented, including each property, slot, method, event, CSS property, and tokens in the component and along with accessibility concerns and/or decisions, linking to accessibility team documentation where applicable.

## Related issue(s)

-   fixes SWC-418

## Screenshots (if appropriate)

---

## Author's checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply.  If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** and **[PULL_REQUESTS](<(https://github.com/adobe/spectrum-web-components/blob/main/PULL_REQUESTS.md)>)** documents.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)
-   [x] <strike>I have added automated tests to cover my changes.</strike> **N/A**
-   [x] <strike>I have included a well-written changeset if my change needs to be published.</strike> **N/A**
-   [x] I have included updated documentation if my change required it.

---

## Reviewer's checklist

-   [ ] Includes a Github Issue with appropriate flag or Jira ticket number without a link
-   [x] <strike>Includes thoughtfully written changeset if changes suggested include `patch`, `minor`, or `major` features</strike> **N/A**
-   [x] <strike>Automated tests cover all use cases and follow best practices for writing</strike> **N/A**
-   [ ] Validated on all supported browsers
-   [x] <strike>All VRTs are approved before the author can update Golden Hash</strike> **N/A**

### Manual review test cases

<!---
    - For the author, please describe in detail what reviewers should test.
    - Include links and manual steps for how the reviewer should go through to verify your changes.
    - Be sure to include all areas of the codebase that might be affected. Any components that use these changes for a dependency should be cross-checked for regressions.
    - For example, changes to Menu Item will affect Picker, Menu, and Action Menu.
-->

-   [ ] _Descriptive Test Statement_

    1. Review [textarea docs](url)
    2. Does the documentation follow the [recommended heading structure](https://opensource.adobe.com/spectrum-web-components/guides/adding-component/#documentation-structure)?
    3. Is the documentation accurate?
    4. Are all the examples accessible?
    5. Do the examples demonstrate how to use the components accessibly?
    6. Are there examples to show how to use the component in context of other components, if necessary?

### Device review

<!--- Verify the above manual tests and visual accuracy utilizing an emulator like Polypane browser or on an actual device. -->

-   [ ] Did it pass in Desktop?
-   [ ] Did it pass in (emulated) Mobile?
-   [ ] Did it pass in (emulated) iPad?

blunteshwar reviewed (commented):
Can you add a sub-section about focus management as well in the accessibility section? This was covered as the last bullet point to the **Keyboard navigation** section so I added a bit more tho the first bullet and renamed the section **Keyboard navigation and focus management**:

```md
#### Keyboard navigation and focus management

- Use <kbd>Tab</kbd> to move focus to and from the textarea
- Use <kbd>Enter</kbd> to create new lines within the textarea
- Use arrow keys to navigate within the text content
- Screen readers announce the textarea label and current value
- Focus indicators are clearly visible for keyboard users
```

cdransf reviewed (approved):
A few comments but nothing blocking. Looks good! ✨

Rajdeepc commented on packages/textfield/textarea.md:
```diff
@@ -7,185 +7,280 @@
+
+- **Text input**: Users can type and edit multiline text
+- **Scrolling**: Content scrolls vertically when it exceeds the visible area
+- **Resizing**: Can be resized by users when the `grows` attribute is not set
```
Can we clarify to the users if this can be disabled or not also?

Rajdeepc commented on packages/textfield/textarea.md:
```diff
@@ -7,185 +7,280 @@
-        Tell us how you are feeling today.
+    <sp-help-text slot="help-text">Words can be space-separated.</sp-help-text>
+    <sp-help-text slot="negative-help-text">
+        Words ust start with the letter "E".
```
```suggestion
        Words must start with the letter "E".
```

Rajdeepc commented on packages/textfield/textarea.md:
```diff
@@ -7,185 +7,280 @@
-## Sizes
+### Anatomy
+
+A text area consistes of an area for the user to enter text and optional placeholder text. It should be used with an [field label](../field-label/index.md) element.
```
```suggestion
A text area consists of an area for the user to enter text and optional placeholder text. It should be used with an [field label](../field-label/index.md) element.
```

Rajdeepc commented on packages/textfield/textarea.md:
```diff
@@ -7,185 +7,280 @@
+<sp-field-label size="xl" for="story-xl">Background</sp-field-label>
 <sp-textfield
     size="xl"
     id="story-0-xl"
```
```suggestion
    id="story-xl"
```

Keeping it consistent with above id patterns

Rajdeepc reviewed (commented):
Some typos, otherwise solid work.

nikkimk commented:
> Can you add a sub-section about focus management as well in the accessibility section?

@blunteshwar This was covered as the last bullet point to the **Keyboard navigation** section so I added a bit more tho the first bullet and renamed the section **Keyboard navigation and focus management**:

```md
#### Keyboard navigation and focus management

- Use <kbd>Tab</kbd> to move focus to and from the textarea
- Use <kbd>Enter</kbd> to create new lines within the textarea
- Use arrow keys to navigate within the text content
- Screen readers announce the textarea label and current value
- Focus indicators are clearly visible for keyboard users
```

blunteshwar reviewed (approved):
Awesome Work!
